ingestion/src/metadata/ingestion/source/database/redshift/connection.py

```diff
@@ -54,65 +55,25 @@ def get_connection(connection: RedshiftConnection) -> Engine:
     )
 
 
-def test_connection(engine: Engine, _) -> TestConnectionResult:
+def test_connection(
+    metadata: OpenMetadata,
+    engine: Engine,
+    service_connection: RedshiftConnection,
+    automation_workflow: Optional[AutomationWorkflow] = None,
+) -> None:
     """
-    Test connection
+    Test connection. This can be executed either as part
+    of a metadata workflow or during an Automation Workflow
     """
-
-    def custom_executor(engine, statement):
-        cursor = engine.execute(statement)
-        return list(cursor.all())
-
-    inspector = inspect(engine)
-    steps = [
-        TestConnectionStep(
-            function=partial(
-                custom_executor,
-                statement=REDSHIFT_GET_DATABASE_NAMES,
-                engine=engine,
-            ),
-            name="Get Databases",
-        ),
-        TestConnectionStep(
-            function=inspector.get_schema_names,
-            name="Get Schemas",
-        ),
-        TestConnectionStep(
-            function=inspector.get_table_names,
-            name="Get Tables",
-        ),
-        TestConnectionStep(
-            function=inspector.get_view_names,
-            name="Get Views",
-            mandatory=False,
-        ),
-        TestConnectionStep(
-            function=partial(
-                custom_executor,
-                statement=REDSHIFT_GET_SVV_TABLE_INFO,
-                engine=engine,
-            ),
-            name="Get Usage and Lineage Svv Table Info",
-            mandatory=False,
-        ),
-        TestConnectionStep(
-            function=partial(
-                custom_executor,
-                statement=REDSHIFT_GET_STL_QUERYTEXT,
-                engine=engine,
-            ),
-            name="Get Usage and Lineage Stl Query Text",
-            mandatory=False,
-        ),
-        TestConnectionStep(
-            function=partial(
-                custom_executor,
-                statement=REDSHIFT_GET_STL_QUERY,
-                engine=engine,
-            ),
-            name="Get Usage and Lineage Stl Query",
-            mandatory=False,
-        ),
-    ]
-
-    return test_connection_db_common(engine, steps)
+    queries = {
+        "GetQueries": REDSHIFT_TEST_QUERY,
+        "GetDatabases": REDSHIFT_GET_DATABASE_NAMES,
+        "GetPartitionTableDetails": REDSHIFT_PARTITION_DETAILS_TEST,
```

Comment:
```suggestion
        "GetQueries": REDSHIFT_TEST_GET_QUERIES,
        "GetDatabases": REDSHIFT_GET_DATABASE_NAMES,
        "GetPartitionTableDetails": REDSHIFT_TEST_GET_PARTITION_DETAILS,
```
I think we need more consistency on naming. wdyt?